### PR TITLE
Docs/radio and checkbox

### PR DESCRIPTION
**Pull request by EdwardIrby**
- Checkbox
- Radio
- Type Signatures column

### CODE REVIEW

**Review by jaketrent — CHANGES_REQUESTED** [review]
imo, `Types.Prop#signature` is what should actually go inside the `type` prop.  Eg:

Instead of:

```
<Types.Prop name="onCheck" type="fucntion" desc="triggers on check toggle" signature="(DOM event, checked, value, name) => ()"/>
```

Do:
```
<Types.Prop name="onCheck" type="(Event, checked: boolean, value: any, name: string) => ()" desc="triggers on check toggle" />
```

What changed:

- Consolidate typing into the `type` prop.
- Use TS types where possible. Eg, `DOM Event` becomes `Event`
- Use names where needed for clarity, otherwise just types: Eg, `Event`

We have done this historically.  I think we keep doing it, and just be more precise by using TS types.

**Comment by EdwardIrby** [comment]
> (Event, checked: boolean, value: any, name: string) => ()

Cool you'll need to go look at some of the other ones the team has done and correct this then as I saw function being used and not the signature.

**Comment by jaketrent** [comment]
yeah, that's fair.  I think we can clean them up.

File: docs/src/pages/components/checkbox.mdx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+route: /components/checkbox
+---
+
+import { Intro, Types, Usage } from '../../components/component'
+
+# Checkbox
+
+## Examples
+### Checked
+Passing the ```checked``` indicates that the checkbox is selected.
```
Comment (jaketrent) [inline docs/src/pages/components/checkbox.mdx:12]:
```suggestion
Passing `checked` indicates that the checkbox is selected.
```

File: docs/src/pages/components/checkbox.mdx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+name: Checkbox
+route: /components/checkbox
+---
+
+import { Intro, Types, Usage } from '../../components/component'
+
+# Checkbox
+
+## Examples
+### Checked
```
Comment (jaketrent) [inline docs/src/pages/components/checkbox.mdx:11]:
```suggestion

### Checked

```

Just stylistic nit, but I'd put some whitespace in throughout.

File: docs/src/pages/components/checkbox.mdx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    <Checkbox checked label="I'm checked" value="Checked-eg" />
+    <Checkbox label="I'm not" value="Checked-eg" />
+  </div>
+)
+
+export default Comp
+
+```
+### Indeterminate
+Passing the ```indeterminate``` flag indicates that the checkbox selection should be obscured as if the control was in a third, indeterminate, state.
```
Comment (jaketrent) [inline docs/src/pages/components/checkbox.mdx:29]:
```suggestion
Passing the `indeterminate` flag indicates that the checkbox selection should be obscured as if the control was in a third, indeterminate, state.
```

File: docs/src/pages/components/checkbox.mdx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    <Checkbox disabled checked label="Checked or not" value="Disabled-eg"/>
+  </div>
+)
+
+export default Comp
+
+```
+
+### Error
+description: Error states are engaged with the ```error``` flag.
```
Comment (jaketrent) [inline docs/src/pages/components/checkbox.mdx:65]:
```suggestion

Error states are engaged with the `error` flag.
```

File: docs/src/pages/components/checkbox.mdx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+```
+
+### In-app
+
+```typescript
+import React from 'react'
+import Checkbox from '@pluralsight/ps-design-system-checkbox'
+import { layout, type, colorsTextIcon, colorsBackgroundLight } from '@pluralsight/ps-design-system-core'
+
+class Comp extends React.Component {
```
Comment (jaketrent) [inline docs/src/pages/components/checkbox.mdx:89]: Could think about doing a functional component rewrite of this to be more modern.

File: docs/src/pages/components/checkbox.mdx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    )
+  }
+}
+export default Comp
+```
+
+## Usage & Types
+
+<Usage install="npm install @pluralsight/ps-design-system-checkbox"
+  import="import Avatar from '@pluralsight/ps-design-system-checkbox'"
```
Comment (jaketrent) [inline docs/src/pages/components/checkbox.mdx:153]:
```suggestion
  import="import Checkbox from '@pluralsight/ps-design-system-checkbox'"
```

File: docs/src/pages/components/dropdown.mdx [hunk 7]
```diff
@@ -381,7 +360,7 @@ export default Example
   <Types.Prop name="error" type="boolean" desc="error state flag"/>
   <Types.Prop name="label" type="string" desc="identifying string for dropdown" />
   <Types.Prop name="menu" type="Dropdown.Item(s)" desc="menu items for dropdown" default="<span></span>" />
-  <Types.Prop name="onChange" type="function" desc="triggered when an item selected ((event, value, label) => ())" />
+  <Types.Prop name="onChange" type="(Event, value, label) => ()" desc="triggered when an item selected ((event, value, label) => ())" />
```
Comment (jaketrent) [inline docs/src/pages/components/dropdown.mdx:363]:
```suggestion
  <Types.Prop name="onChange" type="(Event, value, label) => void" desc="triggered when an item selected" />
```

Just fyi, the old docs use `()` a fair amount in return types.  This is correct in flowtype syntax.  For TS, return types are instead `void`.

File: docs/src/pages/components/radio.mdx [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    <Radio.Button value="advanced" label="Advanced" />
+  </Radio.Group>
+)
+
+export default Comp
+
+```
+
+### Controlled
+Controlled Radios require both ```value``` and ```handleChange``` callback
```
Comment (jaketrent) [inline docs/src/pages/components/radio.mdx:31]:
```suggestion
Controlled `Radio`s require both `value` and `handleChange` callback.
```

File: docs/src/pages/components/radio.mdx [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+)
+
+export default Comp
+
+```
+
+## Usage & Types
+
+<Usage install="npm install @pluralsight/ps-design-system-radio"
+  import="import Avatar from '@pluralsight/ps-design-system-radio'"
```
Comment (jaketrent) [inline docs/src/pages/components/radio.mdx:116]:
```suggestion
  import="import Radio from '@pluralsight/ps-design-system-radio'"
```

File: docs/src/pages/components/checkbox.mdx [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+  packageName="checkbox"
+  />
+
+<Types.Table>
+  <Types.Prop name="checked" type="boolean" desc="marks as selected" default="false"/>
+  <Types.Prop name="disabled" type="boolean" desc="standard input disable flag" default="false"/>
+  <Types.Prop name="error" type="boolean" desc="error state flag" default="false"/>
+  <Types.Prop name="indeterminate" type="boolean" desc="marks as indeterminate" default="false"/>
+  <Types.Prop name="name" type="string" desc="form data identifier"/>
+  <Types.Prop name="label" type="node" desc="display name"/>
```
Comment (jaketrent) [inline docs/src/pages/components/checkbox.mdx:163]:
```suggestion
  <Types.Prop name="label" type="React.ReactNode" desc="display name"/>
```

File: docs/src/pages/components/radio.mdx [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+  <Types.Prop name="name" type="string" desc="form data identifier"/>
+  <Types.Prop name="onChange" type="(Event, value: any) => ()" desc="triggers on radio select"/>
+  <Types.Prop name="subLabel" type="string" desc="supporting text or error messaging"/>
+  <Types.Prop name="value" type="string | number" desc="if Radio.Group is controlled current selected radio value else if uncontrolled initial selected value"/>
+</Types.Table>
+
+### Radio.Button
+
+<Types.Table>
+  <Types.Prop name="label" type="node" desc="display text"/>
```
Comment (jaketrent) [inline docs/src/pages/components/radio.mdx:134]:
```suggestion
  <Types.Prop name="label" type="React.ReactNode" desc="display text"/>
```

**Review by jaketrent — CHANGES_REQUESTED** [review]

**Comment by jaketrent** [comment]
Some of these things are real small nits. But I feel you'd probably want to address at least some of them.  If not, lemme know. We can go forward. I'd be happy to format later.

File: docs/src/pages/components/checkbox.mdx (hunk 5 above)
Comment (EdwardIrby) [reply docs/src/pages/components/checkbox.mdx:89]: Tried that got annoyed when I realized it wasn't a simple 1-1 refactor.  Decided I rather refactor checkbox to work  more like radio in separate PR. Not going to change this one yet as it's outside the scope of the issue.

File: docs/src/pages/components/dropdown.mdx [hunk 12]
```diff
@@ -58,12 +59,13 @@ const Comp = () => (
 )
 export default Comp
 
-===
+```
 
----
```
Comment (jaketrent) [inline docs/src/pages/components/dropdown.mdx:72]: Oh, oops, you had this as a switcher.  

Somehow I may have missed that.  Hence my suggestion to remove the `description: ` prefix on one of these description lines.  

I'm sorry.  I am fine with the switcher, fwiw.

**Review by jaketrent — APPROVED** [review]
Hey Edward, thanks for being open to going the rounds on these.

**Pull request merged**